packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -12,12 +12,187 @@ export interface CodeBuildStartBuildProps extends sfn.TaskStateBaseProps {
    * CodeBuild project to start
    */
   readonly project: codebuild.IProject;
+  /**
+   * Build output artifact settings that override, for this build only.
+   *
+   * @default - the latest ones already defined in the build project.
+   */
+  readonly artifactsOverride?: codebuild.IArtifacts;
+  /**
+   * A buildspec file declaration that overrides, for this build only.
+   *
+   * @default - the latest one already defined in the build project.
+   */
+  readonly buildspecOverride?: string;
+  /**
+   * Contains information that defines how the build project reports the build status to the source provider.
+   *
+   * @default - the latest one already defined in the build project only if the build's source is GitHub, GitHub Enterprise, or Bitbucket.
+   */
+  readonly buildStatusConfigOverride?: codebuild.CfnProject.BuildStatusConfigProperty;
+  /**
+   * A ProjectCache object specified for this build.
+   *
+   * @default - the one defined in the build project.
+   */
+  readonly cacheOverride?: codebuild.CfnProject.ProjectCacheProperty;
+  /**
+   * The name of a certificate for this build.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly certificateOverride?: string;
+  /**
+   * The name of a compute type for this build
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly computeTypeOverride?: codebuild.ComputeType;
+  /**
+   * Specifies if session debugging is enabled for this build.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly debugSessionEnabled?: boolean;
+  /**
+   * The AWS Key Management Service (AWS KMS) customer master key (CMK).
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly encryptionKeyOverride?: string;
+  /**
+   * A container type for this build.
+   *
+   * @default - the one specified in the build project.
+   */
+  readonly environmentTypeOverride?: 'WINDOWS_CONTAINER' | 'LINUX_CONTAINER' | 'LINUX_GPU_CONTAINER'
+  | 'ARM_CONTAINER' | 'WINDOWS_SERVER_2019_CONTAINER';
```

Comment:
is there an existing enum/struct that already contains there? if not, why not create one?